ext/js/pages/settings/dictionary-controller.js

```diff
@@ -1156,7 +1228,12 @@ export class DictionaryController {
         }
 
         await this._deleteDictionary(dictionaryTitle);
-        this._settingsController.trigger('importDictionaryFromUrl', {url: downloadUrl, profilesDictionarySettings});
+        /** @type {Promise<void>} */
+        const importPromise = new Promise((resolve) => {
+            this._settingsController.trigger('importDictionaryFromUrl', {url: downloadUrl, profilesDictionarySettings, onImportDone: resolve});
+        })
```

Comment:
I don't think this is the correct way to do this but this is the only way I can think of to await an event trigger.